internal/manager/run.go

```diff
@@ -189,15 +187,27 @@ func Run(ctx context.Context, c *Config, diagnostic util.ConfigDumpDiagnostic, d
 	}
 
 	if c.Konnect.ConfigSynchronizationEnabled {
+		// In case of failures when building Konnect related objects, we're not returning errors as Konnect is not
+		// considered critical feature, and it should not break the basic functionality of the controller.
+
 		setupLog.Info("Start Konnect client to register runtime instances to Konnect")
-		konnectClient, err := konnect.NewClient(c.Konnect)
+		konnectNodeAPIClient, err := konnect.NewNodeAPIClient(c.Konnect)
+		if err != nil {
+			setupLog.Error(err, "failed creating konnect client, skipping running NodeAgent")
+		} else {
+			hostname, _ := os.Hostname()
+			version := metadata.Release
+			agent := konnect.NewNodeAgent(hostname, version, setupLog, konnectNodeAPIClient)
+			agent.Run()
+		}
+
+		konnectAdminAPIClient, err := adminapi.NewKongClientForKonnectRuntimeGroup(ctx, c.Konnect)
 		if err != nil {
-			return fmt.Errorf("failed to create konnect client: %w", err)
+			setupLog.Error(err, "failed creating Konnect Runtime Group Admin API client, skipping synchronisation")
+		} else {
+			setupLog.Info("Initialized Konnect Admin API client")
+			clientsManager.SetKonnectClient(konnectAdminAPIClient)
 		}
```

Comment:
Let's leave this for now in this PR but as we progress with konnect integration I'd extract this out to a separate function or perhaps even to a separate package.